### ISSUE TITLE
deletion-request ping is missing legacy ID if app is started with data disabled

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#3013]
## Steps to reproduce
- Start app
- Disable data upload
- Force stop app
- Start app
- Trigger Glean Debug Activity uploads: `adb shell am start -n org.mozilla.tv.firefox/mozilla.telemetry.glean.debug.GleanDebugActivity --es tagPings connect`
- Enable data upload
- Disable data upload

### Expected behavior
Legacy client ID appears in deletion-request ping data on dashboard

### Actual behavior
Legacy client ID is not in deletion-request ping. For example:
![image](https://user-images.githubusercontent.com/759372/87365332-d3be8c00-c52a-11ea-9598-c39fe23598ea.png)

### Device information
* Fire TV device: ?
* Latest affected Firefox version: up/master 014894a46c7e764c8e7baa2274e2662d635f0bce
* Earliest affected Firefox version: ?
* Turbo Mode: ?

**Comment by mcomella** [comment]
It's unclear if additional data deletion requests might be sent the next time the app is started but I'd assume not.

The issue is that `UuidMetricType.set` will only set a value if Glean is enabled:
```kotlin
    fun set(value: UUID) {
        if (disabled) {
            return
        }

        // ...
```

**Comment by travis79** [comment]
Okay, the reason this is happening is because when the upload is disabled on the previous run, when the app next starts, `LegacyIds.clientId.set(UUID.fromString(legacyId))` is a no-op for Glean and doesn't get recorded.  In this case, it is probably best to re-set the metric whenever the preference is toggled to "on".

**Comment by abodea** [comment]
Verified as fixed on the latest build **v4.7.1** with Fire Stick TV.
Please note that after following the steps from the description the **Legacy client ID is in deletion-request ping**:

```Kotlin
{
"metadata": {
"geo": {
"city": "Cluj-Napoca",
"subdivision1": "CJ",
"country": "RO",
"db_version": "2020-07-13T19:12:29Z"
},
"isp": {
"organization": "Digi Romania",
"db_version": "2020-07-13T15:10:00Z",
"name": "Digi Romania"
},
"user_agent": {
"os": "Android"
},
"header": {
"date": "Thu, 16 Jul 2020 10:23:06 GMT+00:00",
"x_debug_id": "connect"
},
"document_namespace": "org-mozilla-tv-firefox",
"document_type": "deletion_request",
"document_version": "1"
},
"submission_timestamp": "2020-07-16T10:23:06.588889Z",
"document_id": "c7957502-6083-43bf-805a-d1bfdeaf7bbd",
"normalized_channel": "release",
"normalized_os": "Android",
"normalized_os_version": "5.1.1",
"normalized_country_code": "RO",
"sample_id": 51,
"client_info": {
"android_sdk_version": "22",
"app_build": "41970208",
"app_channel": "release",
"app_display_version": "4.7.1",
"architecture": "armeabi-v7a",
"client_id": "b1141e80-2dac-4951-b4cb-d6a4f0ac63b5",
"device_manufacturer": "Amazon",
"device_model": "AFTT",
"first_run_date": "2020-07-16+03:00",
"os": "Android",
"os_version": "5.1.1",
"telemetry_sdk_build": "21.3.0"
},
"metrics": {
"uuid": {
"legacy.ids.client_id": "fe24b3c3-e210-41f8-abdf-0d7d4f5c921d"
}
},
"ping_info": {
"end_time": "2020-07-16T13:23+03:00",
"ping_type": "deletion_request",
"seq": 0,
"start_time": "2020-07-16T13:20+03:00"
}
}
```
